core/mount/mount_idmapped_utils_linux_go122.go

```diff
@@ -0,0 +1,88 @@
+//go:build !go1.23 && linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mount
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/containerd/containerd/v2/pkg/sys"
+
+	"golang.org/x/sys/unix"
+)
+
+// getUsernsFD returns pinnable user namespace's file descriptor.
+func getUsernsFD(uidMaps, gidMaps []syscall.SysProcIDMap) (*os.File, error) {
+	var pidfd int
+
+	proc, err := os.StartProcess("/proc/self/exe", []string{"containerd[getUsernsFD]"}, &os.ProcAttr{
+		Sys: &syscall.SysProcAttr{
+			Cloneflags:  unix.CLONE_NEWUSER,
+			UidMappings: uidMaps,
+			GidMappings: gidMaps,
+			// NOTE: It's reexec but it's not heavy because subprocess
+			// be in PTRACE_TRACEME mode before performing execve.
+			Ptrace:    true,
+			Pdeathsig: syscall.SIGKILL,
```

Comment:
Just curious, SIGKILL does kill the process if it is using `Ptrace: true`?

Reply:
Pdeathsig is used to kill the tracee if the parent process, like containerd, exits. So it won't cause leaky subprocess.

Reply:
Yes, I understand that. My question is: if the process is waiting with `Ptrace: true`, will it react to SIGKILL?

Reply:
Oh. Yes. The tracee is in STOP status so kernel can kill it.

Reply:
Perfect, thanks!